From 1533169b74a3f5fec8d441eec547fe1cf85bac06 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Wed, 25 Jun 2014 09:25:33 +0100
Subject: [PATCH 235/351] switch: arizona: Reset accessory detection mode to
 default

For best performance the accessory detection mode should be left in its
default state when no detection is being performed.

Change-Id: I323591366e68fa590460ee304800058562e734b3
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 44b64a7..83c2071 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -792,6 +792,11 @@ void arizona_hpdet_stop(struct arizona_extcon_info *info)
 			   ARIZONA_HP_IMPEDANCE_RANGE_MASK |
 			   ARIZONA_HP_POLL, 0);
 
+	/* Reset to default mode */
+	regmap_update_bits(arizona->regmap,
+			   ARIZONA_ACCESSORY_DETECT_MODE_1,
+			   ARIZONA_ACCDET_MODE_MASK, 0);
+
 	arizona_extcon_do_magic(info, 0);
 
 	pm_runtime_put_autosuspend(info->dev);
@@ -924,6 +929,11 @@ void arizona_micd_stop(struct arizona_extcon_info *info)
 		mutex_unlock(&arizona->reg_setting_lock);
 	}
 
+	/* Reset to default mode */
+	regmap_update_bits(arizona->regmap,
+			   ARIZONA_ACCESSORY_DETECT_MODE_1,
+			   ARIZONA_ACCDET_MODE_MASK, 0);
+
 	regulator_disable(info->micvdd);
 
 	pm_runtime_mark_last_busy(info->dev);
-- 
1.7.9.5

